bookserver/session.py

```diff
@@ -41,3 +43,13 @@ async def load_user(user_id: str):
     """
     rslogger.debug(f"Going to fetch {user_id}")
     return await fetch_user(user_id)
+
+
+async def is_instructor(request: Request) -> bool:
+    user = request.state.user
+    if user is None:
+        raise HTTPException(401)
+    elif len(await fetch_instructor_courses(user.id, user.course_id)) > 0:
```

Comment:
Shouldn't be check that the user is an instructor for the provided course?

Reply:
That is exactly what this is doing.

Reply:
Oh, duh, I missed the second parameter.